README.md

```diff
@@ -51,214 +122,232 @@ where
 
-- In order to set these configurations manually, edit `demo_config.h` in `demos/mqtt/mqtt_demo_mutual_auth/` and `demos/http/http_demo_mutual_auth/` to `#define` the following:
+### v4_beta_deprecated branch (formerly named v4_beta)
 
-	- Set `AWS_IOT_ENDPOINT` to your custom endpoint. This is found on the *Settings* page of the AWS IoT Console and has a format of `ABCDEFG1234567.iot.us-east-2.amazonaws.com`.
+The [v4_beta_deprecated](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v4_beta_deprecated) branch contains a beta version of the C-SDK libraries, which is now deprecated. This branch was earlier named as v4_beta, and was renamed to v4_beta_deprecated. The libraries in this branch will not be released. However, critical bugs will be fixed and tested. No new features will be added to this branch.
 
-	- Set `ROOT_CA_CERT_PATH` to the path of the root CA certificate downloaded when setting up the device certificate in [AWS IoT Account Setup](#aws-iot-account-setup).
+## Getting Started
 
-	- Set `CLIENT_CERT_PATH` to the path of the client certificate downloaded when setting up the device certificate in [AWS IoT Account Setup](#aws-iot-account-setup).
+### Cloning
 
-	- Set `CLIENT_PRIVATE_KEY_PATH` to the path of the private key downloaded when setting up the device certificate in [AWS IoT Account Setup](#aws-iot-account-setup).
+This repository uses [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) to bring in the C-SDK libraries (eg, MQTT ) and third-party dependencies (eg, mbedtls for POSIX platform transport layer).
+Note: If you download the ZIP file provided by GitHub UI, you will not get the contents of the submodules (The ZIP file is also not a valid git repository). If you download from the [202011.00 Release Page](https://github.com/aws/aws-iot-device-sdk-embedded-C/releases/tag/202011.00) page, you will get the entire repository (including the submodules) in the ZIP file, aws-iot-device-sdk-embedded-c-202011.00.zip.
+To clone using HTTPS:
 
+```sh
+git clone --recurse-submodules https://github.com/aws/aws-iot-device-sdk-embedded-C.git
+```
 
-### Build Steps
+Using SSH:
 
-1. Go to the root directory of this repository.
+```sh
+git clone --recurse-submodules git@github.com:aws/aws-iot-device-sdk-embedded-C.git
+```
 
-1. Create build directory: `mkdir build && cd build`
+If you have downloaded the repo without using the `--recurse-submodules` argument, you need to run:
 
-1. Run *cmake* while inside build directory: `cmake ..`
+```sh
+git submodule update --init --recursive
+```
 
-1. Run this command to build the demos: `make`
+### Building and Running Demos
```

Comment:
There are no instructions to use the S3 demos? Like how to use the script to generate a presigned URL and pass that as a CMake flag.